### ISSUE TITLE
Intermittent failure Bootstrapping internal Streams

### ISSUE THREAD

**Issue by maddisondavid** [pravega/pravega#1491]
**Problem description**
The controller sometimes fails when attempting to Bootstrap the controller event processors streams. 

As well as the issue with the controller state not being correct, another problem is that the SegmentStore just keeps retrying to connect to these internal streams BUT continues to process stream requests.  From the outside the SegmentStore behaves correctly but scaling events will never happen. (this was all spotted on a Scaling test)

**Problem location**
ControllerEventProcessors::Bootstrap

The following logs show a single controller and segment store startup.  The segment store is started first.

We see the attempt at bootstrapping :
```
2017-06-27 17:44:05,478 23197 [ControllerServiceStarter STARTING] INFO  i.p.c.s.e.ControllerEventProcessors - Bootstrapping controller event processors
```
But this is quickly followed by WireFailures
```
2017-06-27 17:44:32,179 49898 [epollEventLoopGroup-2-2] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:32,179 49898 [epollEventLoopGroup-2-1] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:32,278 49997 [epollEventLoopGroup-2-4] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:32,278 49997 [epollEventLoopGroup-2-5] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:32,278 49997 [epollEventLoopGroup-2-3] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:33,480 51199 [ControllerServiceStarter STARTING] INFO  i.p.c.s.ControllerServiceStarter - Awaiting start of REST server
2017-06-27 17:44:33,582 51301 [RESTServer STARTING] INFO  i.p.c.server.rest.RESTServer - Starting REST server listening on port: 10080
2017-06-27 17:44:39,580 57299 [epollEventLoopGroup-2-8] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
2017-06-27 17:44:39,580 57299 [epollEventLoopGroup-2-2] INFO  i.p.s.p.netty.FailingReplyProcessor - Received hello: WireCommands.Hello(type=HELLO, highVersion=1, lowVersion=1)
```

This is followed by the stream create failing
```
2017-06-27 17:45:10,078 87797 [epollEventLoopGroup-2-6] WARN  i.p.c.t.Stream.StreamMetadataTasks - Create stream failed due to 
java.util.concurrent.CompletionException: io.pravega.common.util.RetriesExhaustedException: java.util.concurrent.CompletionException: io.pravega.controller.server.WireCommandFailedException: WireCommandFailed with type CREATE_SEGMENT reason ConnectionDropped
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:647)
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632)
```

Looking at the stream metadata in ZK the stream is shown stuck in  `CREATING`
```
get /pravega/pravega-cluster/store/_system/_requeststream/state
��~r/io.pravega.controller.store.stream.tables.Statexrjava.lang.EnumxpCREATING
```

[stdout-SegmentStore.txt](https://github.com/pravega/pravega/files/1108043/stdout-SegmentStore.txt)
[stdout-Controller.txt](https://github.com/pravega/pravega/files/1108044/stdout-Controller.txt)

**Suggestions for an improvement**
Fix issue with Controller startup

SegmentStore should not be classed as started and processing stream requests until it has successfully connected to the internal streams, otherwise the Segment store is in an incomplete state and scaling events will fail invisibly.

**Comment by chrisdail** [comment]
This is a pretty major issue. There is a high probability when creating a pravega environment that you will be unable to create streams. Client code error is this:

```
! Causing: java.lang.RuntimeException: io.pravega.client.stream.impl.ControllerFailureException: Failed to create stream: StreamConfiguration(scope=hulk, streamName=small, scalingPolicy=ScalingPolicy(type=FIXED_NUM_SEGMENTS, targetRate=0, scaleFactor=0, minNumSegments=3), retentionPolicy=RetentionPolicy(type=SIZE, value=53687091200))
! at io.pravega.common.concurrent.FutureHelpers.getAndHandleExceptions(FutureHelpers.java:166)
! at io.pravega.client.admin.impl.StreamManagerImpl.createStream(StreamManagerImpl.java:41)
! at com.emc.nautilus.pravega.utils.PravegaStreamHelper.createStream(PravegaStreamHelper.java:52)
! at com.emc.nautilus.pravega.master.scheduler.PravegaTestPreparer.createStreams(PravegaTestPreparer.java:75)
```

**Comment by maddisondavid** [comment]
Our test environment hit this issue again which is running a build that has this fix applied (commit e5e9896)

[1491-Controller.txt](https://github.com/pravega/pravega/files/1121569/1491-Controller.txt)
[1491-Segment-Store.txt](https://github.com/pravega/pravega/files/1121570/1491-Segment-Store.txt)
